docs/account/user-session-management.md

```diff
@@ -0,0 +1,11 @@
+# User session management
+
+To reduce the risk of unauthorized access to your account, Codacy sets automatic expiration timeouts for every session. When a session expires, Codacy invalidates the session both on the client and server side.<!--TODO PLUTO-883 Validate the behavior when there is an active session on the Git provider side, would that avoid an explicit re-login on refresh-->
```

Comment:
@andrzej-janczak When Codacy invalidates the session (client and server side), what happens if the Git provider session is still valid? 🤔 
Will the user be automatically logged in again after refreshing the page? Or will Codacy redirect the user to the login page, and the user must choose the provider again?

Also, is there any other technical detail that users must be aware of and worth including in the product documentation?

Reply:
So we have 2 main scenarios here that depends on if browser/user have cookie `codacy.entity.provider` 

if yes then: 
No user action will be required. Everything will happen by redirects. This will be "invisible" for user.  
1. no cookie OR cookie expired-> session invalidated 
2. Autologin mechanism apply - having the `codacy.entity.provider` we know which provider we should login with. 
3. We prefrom /auth action, this do redirect to provider and back again 
4. User is logged in 

if NO: 
Same as above but before that user need to choose provider on login page.

Reply:
Hey @andrzej-janczak! Thank you so much for these details. 🙇 
I have described both scenarios without including too much information about that cookie. [Have a look](https://github.com/codacy/docs/pull/2062/files#diff-8b03676a3f275e4d112f69e2c44f2bb2a433e11b692e05b37647859ec4d40905R5).